biz.aQute.bndlib/src/aQute/bnd/osgi/MultiReleaseJars.java

```diff
@@ -0,0 +1,112 @@
+package aQute.bnd.osgi;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import aQute.bnd.exceptions.Exceptions;
+
+/**
+ * Multi Release jars are another error magnet extension to Java. Instead of
+ * having the flat class space of Java and Javac, it allows developers to put
+ * multiple versions of a class in versioned directory.
+ * <p>
+ *
+ * @see "https://docs.oracle.com/en/java/javase/19/docs/specs/jar/jar.html#multi-release-jar-files"
+ *      <p>
+ */
+public class MultiReleaseJars {
+	final static Pattern VERSIONED_P = Pattern.compile("META-INF/versions/(?<release>\\d+)/(?<path>.*)");
+
+	/**
+	 * Check if a JAR is a multi release jar.
+	 *
+	 * @param jar the jar
+	 * @return true if this is a multi release jar
+	 */
+	public static boolean isMulti(Jar jar) {
+		try {
+			Domain domain = Domain.domain(jar.getManifest());
+			return Processor.isTrue(domain.get(Constants.MULTI_RElEASE));
+		} catch (Exception e) {
+			throw Exceptions.duck(e);
```

Comment:
In that case it is most probably not even a jar at all and one can return false.